### ISSUE TITLE
Same trait in return impl trait should warn

### ISSUE THREAD

**Issue by DutchGhost** [rust-lang/rust#54328]
Having the same trait specified in the list of traits a function returns, does not give a warning:

```Rust
fn example() -> impl Copy + Copy + Copy {
    0
}

fn main() {
    example();
}
```

It should perhaps give a warning like:
```
warning: found multiple identical traits: `fn example() -> impl Copy + Copy + Copy`
```

**Comment by DutchGhost** [comment]
EDIT:

dyn Trait does not give a warning as well:

```Rust
fn dyn_trait() -> Box<dyn Send + Send> {
    Box::new(())
}

fn impl_trait() -> impl Send + Send  { }

fn main() {
    dyn_trait();
    impl_trait();
}
```

**Comment by rkruppe** [comment]
This isn't limited to `impl` or `dyn`, you can have redundant bounds everywhere there are bounds.

**Comment by estebank** [comment]
If we were to add a lint it should explicitly ignore macro contexts.
